firewood/src/v2/api.rs

```diff
@@ -8,9 +8,9 @@ use async_trait::async_trait;
 /// A `KeyType` is something that can be xcast to a u8 reference,
 /// and can be sent and shared across threads. References with
 /// lifetimes are not allowed (hence 'static)
-pub trait KeyType: AsRef<[u8]> + Send + Sync + Debug + 'static {}
+pub trait KeyType: AsRef<[u8]> + Send + Sync + Debug {}
```

Comment:
This is going to break the ownership of keys in a number of places. Maybe instead rename this to OwnedKey and add:
```
pub trait Key: AsRef<[u8]> + Send + Sync + Debug {}
```

Sprinkle `Key` in where it's clear we should be borrowing.

Reply:
It's not breaking anything. `KeyType + 'static` is a subset of `KeyType` and should be added where necessary, not the other way around.

Reply:
OK that makes sense. Do we need to add any `'static`s anywhere?